sdk/src/system_instruction.rs

```diff
@@ -147,7 +147,7 @@ pub fn create_account(
     program_id: &Pubkey,
 ) -> Instruction {
     let account_metas = vec![
-        AccountMeta::new(*from_pubkey, true),
+        AccountMeta::new(*from_pubkey, lamports != 0), // no signature required if no transfer
```

Comment:
What's the point of this feature exactly? And if you use it, why require `from_pubkey`?

Reply:
I could round out system program with another set of functions/instructions that don't require a `from`, but I think they'd only manifest as `system_instruction.rs` entry points, as so many things funnel down.

My next PR makes clear the value of this feature.  Stay tuned!

Reply:
Something still smells funny here. Is it possible that `CreateAccount` shouldn't have `from` or `lamports` parameters at all?

Reply:
your new issue nicely describes what I'm going for here

Reply:
Good to hear!  So baby steps, how about we leave `CreateAccount` alone and instead add the proposed `Allocate` instruction in this PR?